artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/amq/AMQConsumer.java

```diff
@@ -232,11 +232,24 @@ public ConsumerId getId() {
       return info.getConsumerId();
    }
 
-   public void acquireCredit(int n) {
+   public void acquireCredit(int n, boolean delivered) {
       if (messagePullHandler.get() != null) {
          //don't acquire any credits when the pull handler controls it!!
          return;
       }
+
+      if (delivered) {
+         deliveredAcksCreditExtension += n;
+      } else if (deliveredAcksCreditExtension > 0) {
```

Comment:
that is better, now the normal deliveredAcksCreditExtension == 0 branch is clear